src/StreamJsonRpc/MessagePackFormatter.cs

```diff
@@ -2285,7 +2290,31 @@ internal void SetTopLevelProperty<T>(string name, [MaybeNull] T value)
 
         [DebuggerDisplay("{" + nameof(DebuggerDisplay) + ",nq}")]
         [DataContract]
-        private class JsonRpcRequest : Protocol.JsonRpcRequest, IJsonRpcMessageBufferManager, IJsonRpcMessagePackRetention
+        private class OutboundJsonRpcRequest : Protocol.JsonRpcRequest, IMessageWithTopLevelPropertyBag
+        {
+            private readonly MessagePackFormatter formatter;
+
+            internal OutboundJsonRpcRequest(MessagePackFormatter formatter)
+            {
+                this.formatter = formatter ?? throw new ArgumentNullException(nameof(formatter));
+            }
+
+            public TopLevelPropertyBag? TopLevelPropertyBag { get; set; }
+
+            public override bool TrySetTopLevelProperty<T>(string name, [MaybeNull] T value)
```

Comment:
Same comment as the Json requests messages here regarding TrySetTopLevelProperty implementation in the inbound one

Reply:
Same response, but especially true here, since remote RPC target tests don't run on messagepack at all.